### PR TITLE
Implement full TOC for `docs/Reference`

### PR DESCRIPTION
**Pull request by jsumners**
This PR finishes the `docs/Reference/index.md` file. It also applies hard line wrapping to each of the reference documents. 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: docs/Reference/index.md [hunk 1]
```diff
@@ -1,8 +1,71 @@
...
+For the full table of contents (TOC), see [below](#reference-toc). The following
+list is a subset of the full TOC that detail core Fastify APIs and concepts in
+order of most likely importance to the reader:
 
 + [Server](./Server.md): Documents the core Fastify API. Includes documentation
 for the factory function and the object returned by the factory function.
++ [Lifecycle](./Lifecycle.md): Explains the Fastify request lifecycle and
+illustrates where [Hooks](./Hooks.md) are available for integrating with it.
++ [Routes](./Routes.md): Details how to register routes with Fastify and how
+Fastify builds and evaluates the routing trie.
```
Comment (Fdawgs) [inline docs/Reference/index.md:15]:
```suggestion
Fastify builds and evaluates the routing tree.
```

File: docs/Reference/index.md [hunk 2]
```diff
@@ -1,8 +1,71 @@
...
++ [Decorators](./Decorators.md): Explains the server, request, and response
+decorator APIs.
++ [Hooks](./Hooks.md): Details the API by which Fastify plugins can inject
+themselves into Fastify's handling of the request lifecycle.
+
+
+## Reference Documentation Table Of Contents
+<a id="reference-toc"></a>
+
+This table of contents is in alphabetical order.
```
Comment (Fdawgs) [inline docs/Reference/index.md:35]:
```suggestion
This table of contents is in ascending alphabetical order.
```

File: docs/Reference/index.md [hunk 3]
```diff
@@ -1,8 +1,71 @@
 <h1 align="center">Fastify</h1>
 
-## Reference Documentation Table Of Contents
+## Core Documents
+<a id="reference-core-docs"></a>
+
+For the full table of contents (TOC), see [below](#reference-toc). The following
+list is a subset of the full TOC that detail core Fastify APIs and concepts in
+order of most likely importance to the reader:
```
Comment (Fdawgs) [inline docs/Reference/index.md:8]: Not 100% on this line, I don't know if "assumed importance" would read better?

**Review by Fdawgs — CHANGES_REQUESTED** [review]

File: docs/Reference/index.md (hunk 1 above)
Comment (jsumners) [reply docs/Reference/index.md:15]: https://en.m.wikipedia.org/wiki/Trie

File: docs/Reference/index.md (hunk 2 above)
Comment (jsumners) [reply docs/Reference/index.md:35]: That is redundant.

File: docs/Reference/index.md (hunk 1 above)
Comment (Fdawgs) [reply docs/Reference/index.md:15]: Well I've learned something new today!

**Review by Fdawgs — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

File: docs/Reference/index.md (hunk 3 above)
Comment (jsumners) [reply docs/Reference/index.md:8]: I'm going to go ahead and merge this. If your wife would like to tie break this one, or we can think of something better, let's PR a change.

**Pull request merged**